### PR TITLE
WIP: Driver for Bebop 2 ultrasonic rangefinder

### PR DESCRIPTION
**Pull request by eyeam3**
This PR adds a driver for the ultrasonic rangefinder on Parrot Bebop 2. It includes a basic IIO implementation and the sensor interfacing.

The driver is basically finished but marked as WIP because it was not tested in flight due to integration issues in the PX4 firmware. See PX4/Firmware#6126.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@lfelipe @lucasdemarchi Given you worked on IIO before it would be great to sync on how we can collaborate also on the Linux low-level driver side more closely.

File: framework/src/SPIDevObj.cpp [hunk 1]
```diff
@@ -271,7 +282,7 @@ int SPIDevObj::_modifyReg(uint8_t address, uint8_t clearbits, uint8_t setbits)
 
 int SPIDevObj::bulkRead(DevHandle &h, uint8_t address, uint8_t *out_buffer, int length)
 {
-#if defined(__DF_RPI) || defined(__DF_EDISON)
+#if defined(__DF_RPI) || defined(__DF_EDISON) || defined(__DF_BEBOP)
```
Comment (LorenzMeier) [inline framework/src/SPIDevObj.cpp]: @mcharleb This is architecturally horrible - could you look into how we can restructure this to not use ifdefs?

**Comment by LorenzMeier** [comment]
@eyeam3 Very cool! My comments are just generally on DF - I'm hunting down every ifdef I can. We need to drive those out of the codebase. Not necessarily in this PR though, so this should not gate it going in.

File: framework/src/SPIDevObj.cpp (hunk 1 above)
Comment (julianoes) [inline framework/src/SPIDevObj.cpp]: We could it improve by

```
#if defined (__DF_QURT)
// do it the QuRT qay
#else
// do it the usual way
#endif
```

File: framework/src/SPIDevObj.cpp [hunk 2]
```diff
@@ -425,13 +436,17 @@ int SPIDevObj::setBusFrequency(DevHandle &h, SPI_FREQUENCY freq_hz)
 
 int SPIDevObj::_setBusFrequency(SPI_FREQUENCY freq_hz)
 {
-#if defined(__DF_RPI)
+#if defined(__DF_RPI) || defined(__DF_BEBOP)
 
 	/* implement sensor interface via rpi spi */
 	// RPI rounds down freq_hz to powers of 2
 	// Speeds available: 0.5, 1, 2, 4, 8, 16, and 32 MHz
```
Comment (julianoes) [inline framework/src/SPIDevObj.cpp]: The comment is now outdated, or even was before.

**Pull request merged**